telegram/ext/dispatcher.py

```diff
@@ -229,30 +236,89 @@ def _pooled(self):
                 break
 
             promise.run()
+
+            if not promise.exception:
+                self.update_persistence(update=promise.update)
+                continue
+
             if isinstance(promise.exception, DispatcherHandlerStop):
                 self.logger.warning(
                     'DispatcherHandlerStop is not supported with async functions; func: %s',
                     promise.pooled_function.__name__)
+                continue
+
+            if promise.pooled_function in self.error_handlers:
+                self.logger.error('An error was raised while processing the update '
+                                  'and an uncaught error was raised while handling '
+                                  'the error with an error_handler')
```

Comment:
@Bibo-Joshi quoting you (as the original discussion is marked "Outdated":
> return statement: good idea. wording: I just used the wording that we use everywhere, i.e. also in process_update and in JobQueue … Should I change everywhere?

---
In this context (and all the locations it is repeated down in this function) I find the error text to be too long. I prefer the shortest sentence possible which pass the idea.

In this case saying:
`An uncaught error was raised by an error_handler` is still 100% correct and pass the idea.
It's obvious that the error handler wouldn't be called unless there was an error, right?
So why explicitly say that it happened? It just clutters the log (of the user) and makes our code unreadable.

```suggestion
                self.logger.error('An uncaught error was raised by an error_handler')
```

Regarding changing the log text in other places, it doesn't have to happen now. Make sure that the log you issue now makes sense, and if you want open a new issue to fix other logs in the future.